flink/v1.19/flink/src/main/java/org/apache/iceberg/flink/maintenance/operator/JdbcLockFactory.java

```diff
@@ -0,0 +1,246 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.flink.maintenance.operator;
+
+import java.io.IOException;
+import java.sql.DatabaseMetaData;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.SQLNonTransientConnectionException;
+import java.sql.SQLTimeoutException;
+import java.sql.SQLTransientConnectionException;
+import java.util.Map;
+import java.util.UUID;
+import org.apache.iceberg.jdbc.JdbcClientPool;
+import org.apache.iceberg.jdbc.UncheckedInterruptedException;
+import org.apache.iceberg.jdbc.UncheckedSQLException;
+import org.apache.iceberg.relocated.com.google.common.annotations.VisibleForTesting;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.util.PropertyUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class JdbcLockFactory implements TriggerLockFactory {
+  private static final Logger LOG = LoggerFactory.getLogger(JdbcLockFactory.class);
+  private static final String INIT_LOCK_TABLES_PROPERTY = "maintenance.lock.jdbc.init-lock-tables";
+  private static final String LOCK_TABLE_NAME = "maintenance_lock";
+  private static final int MAX_PREFIX_LENGTH = 100;
+  private static final String CREATE_LOCK_TABLE_SQL =
+      "CREATE TABLE "
+          + LOCK_TABLE_NAME
+          + "(LOCK_TYPE CHAR(1) NOT NULL, ISSUER VARCHAR("
+          + MAX_PREFIX_LENGTH
```

Comment:
I looked at the `LockManager` class and think maybe we can learn from it. I like the naming of `entityId` (instead of `issuer`) and `ownerId` (instead of `lockId`).

Reply:
I think that would be misleading. The entityId is more-or-less descriptive, but the ownerId is different, as this lock doesn't really care about the ownership at all.

I opted for renaming them differently, which might be better:
- LOCK_TYPE
- LOCK_ID - previously `ISSUER`
- INSTANCE_ID - previously `LOCK_ID`

Reply:
agree that `OWNER_ID` is misleading now I can remember the purpose of this field after seeing the code comment on the race condition, which helped me refresh my memory.

current naming LGTM.